File: TF10. METHODS. DEB AND TROUBLESHOOTING CODE - EX (Java)/P06_PrimeChecker.java
import java.util.Scanner;

public class P06_PrimeChecker {
    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);

        long num = Long.parseLong(in.nextLine());

        System.out.println(CheckPrime(num));
    }

    private static String CheckPrime(long n) {

        if (n <= 1) {
            return "False";
        }
        if (n == 2) {
            return "True";
        }
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return "False";
            }
        }
        return "True";
    }
}

File: TF10. METHODS. DEB AND TROUBLESHOOTING CODE - EX/P06.PrimeChecker.cs
using System;

namespace P06.PrimeChecker
{
    class Program
    {
        static void Main(string[] args)
        {
            long num = long.Parse(Console.ReadLine());

            Console.WriteLine(CheckPrime(num));
        }

        private static bool CheckPrime(long n)
        {
            if (n <= 1)
            {
                return false;
            }
            if (n == 2)
            {
                return true;
            }
            for (int i = 2; i <= Math.Sqrt(n); i++)
            {
                if (n % i == 0)
                {
                    return false;
                }
            }
            return true;
        }
    }
}
